pmd-core/src/main/java/net/sourceforge/pmd/rules/RuleBuilder.java

```diff
@@ -173,7 +175,7 @@ private void throwUnknownLanguageVersionException(String minOrMax, String unknow
                                            + "' for Language '" + language.getTerseName()
                                            + "' for Rule " + name
                                            + "; supported Language Versions are: "
-                                           + LanguageRegistry.commaSeparatedTerseNamesForLanguageVersion(language.getVersions()));
+                                           + language.getVersions().stream().map(LanguageVersion::getTerseName).collect(Collectors.joining(", ")));
```

Comment:
I just realize this message was incorrect.

If due to a typo a user's rule is configured as:

```xml
<rule name="MyRule" language="java" minimumLanguageVersion="99">
```

We are printing something such as:

> Unknown minimum Language Version 99 for Language 'java' for Rule MyRule; supported Language Versions are: java 1.3, java 1.4, java 1.5, java 1.6, java 1.7, java 1.8, java 9, …

So, a user would understand, that instead of `99`, they should write `java 9` as:

```xml
<rule name="MyRule" language="java" minimumLanguageVersion="java 9">
```

which would not be valid.

We should probably just do:

```java
language.getVersions().stream().map(LanguageVersion::getVersion).collect(Collectors.joining(", ")));
```

to produce:

> Unknown minimum Language Version 99 for Language 'java' for Rule MyRule; supported Language Versions are: 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 9, …

Reply:
Fixed